### PR TITLE
Remove DoFHandlerType from internal::DoFHandlerImplementation::Iterators

### PR DESCRIPTION
**Pull request by peterrum**
references #10333

### CODE REVIEW

File: include/deal.II/dofs/dof_iterator_selector.h [hunk 1]
```diff
@@ -56,10 +59,10 @@ namespace internal
      * @ref Iterators
      * module for more information.
      */
-    template <template <int, int> class DoFHandlerType, int spacedim, bool lda>
-    struct Iterators<DoFHandlerType<1, spacedim>, lda>
+    template <int spacedim, bool lda>
+    struct Iterators<1, spacedim, lda>
     {
-      using DoFHandler_type = DoFHandlerType<1, spacedim>;
+      using DoFHandler_type = DoFHandler<1, spacedim>;
```
Comment (drwells) [inline include/deal.II/dofs/dof_iterator_selector.h:65]: Would you mind making these each `DoFHandlerType`?

Comment (drwells) [reply include/deal.II/dofs/dof_iterator_selector.h:65]: On second thought - could we just deprecate these? They aren't used anywhere in the library AFAICT.

Comment (peterrum) [reply include/deal.II/dofs/dof_iterator_selector.h]: Good catch!

**Comment by drwells** [comment]
I didn't realize they were in `namespace internal` - removing them is even better :+1:

**Comment by peterrum** [comment]
> I didn't realize they were in namespace internal - removing them is even better +1

Internal namespace is the friend of us as developers :P

**Pull request merged**